### ISSUE TITLE
Nested complex type issue

### ISSUE THREAD

**Issue by ysmoradi** [simple-odata-client/Simple.OData.Client#617]
I've attached a sample repo. It has 3 automated integration tests. All those tests are calling the same odata action which accepts a nested complex type. Tests which are using http client are able to call the api but odata client fails to send a request due following exception:

```
Microsoft.OData.ODataException: The parameter 'ComplexType2' is of Edm type kind 'Complex'. You cannot call WriteValue on a parameter that is not of Edm type kinds 'Primitive', 'Enum' or 'Complex'.
```

You can also run the asp.net app, it opens swagger page for you. You can also take a look at $metadata by opening http://localhost:5000/odata/MyApp/$metadata

I'm using the latest version.

[SimpleODataClientNestedComplexTypeIssue.zip](https://github.com/simple-odata-client/Simple.OData.Client/files/3170348/SimpleODataClientNestedComplexTypeIssue.zip)

Thanks for your fantastic library (-:

**Comment by ysmoradi** [comment]
Any news on this?

**Comment by mike9182** [comment]
This may have been fixed as part of pull request #634.

**Comment by ysmoradi** [comment]
> This may have been fixed as part of pull request #634.

No, it's not. Check my PR #715

**Comment by object** [comment]
Released in 5.11.1.
